Toolkit/ArcGISToolkit/AR/ArcGISARView.swift

```diff
@@ -0,0 +1,682 @@
+        return false
+    }
+    
+    /// Resets the device tracking using a spacial anchor
+    ///
+    /// - Returns: reset operation success or failure
+    public func resetUsingSpatialAnchor() -> Bool {
+        return false
+    }
+    
+    /// Starts device tracking
+    public func startTracking() {
+        
+        if !isSupported {
+            didStartOrFailWithError(ArcGISARView.notSupportedError())
+            return
+        }
+        
+        if let origin = originCamera {
+            //
+            // we have a starting camera
+            initialTransformationMatrix = origin.transformationMatrix
+            sceneView.setViewpointCamera(origin)
+            finalizeStart()
+        }
+        else {
+            //
+            // no starting camera, use location manger to get initial location
+            let authStatus = CLLocationManager.authorizationStatus()
+            switch authStatus {
+            case .notDetermined:
+                startWithAccessNotDetermined()
+            case .restricted, .denied:
+                startWithAccessDenied()
+            case .authorizedAlways, .authorizedWhenInUse:
+                startWithAccessAuthorized()
+            }
+        }
+        
+        //
+        // we need to know when the device orientation changes in order to update the Camera transformation
+        UIDevice.current.beginGeneratingDeviceOrientationNotifications()
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(self.orientationChanged(notification:)),
+            name: UIDevice.orientationDidChangeNotification,
+            object: nil
+        )
+        
+        //
+        // reset frameCount and start timer to capture frame rate
+        frameCount = 0
+        frameCountTimer = Timer.scheduledTimer(withTimeInterval: 1.0, repeats: true, block: { [weak self] (timer) in
+            print("Frame rate = \(String(reflecting: self?.frameCount))")
+            self?.frameCount = 0
+        })
+    }
+
+    /// Suspends device tracking
+    public func stopTracking() {
+        arSCNView.session.pause()
+        stopUpdatingLocationAndHeading()
+        
+        UIDevice.current.endGeneratingDeviceOrientationNotifications()
+        NotificationCenter.default.removeObserver(self)
+
+        frameCountTimer?.invalidate()
+    }
+    
+    // MARK: Private
+    
+    /// Operations that happen after device tracking has started
+    fileprivate func finalizeStart() {
+        //
+        // hide the camera image if necessary
+        arSCNView.isHidden = !renderVideoFeed
+        
+        //
+        // run the ARSession
+        arSCNView.session.run(arConfiguration, options:.resetTracking)
+        didStartOrFailWithError(nil)
+    }
+
+    /// Adds subView to superView with appropriate constraints
+    ///
+    /// - Parameter subview: the subView to add
+    fileprivate func addSubviewWithConstraints(_ subview: UIView) {
```

Comment:
Is there a preferred way to do this?  I thought Autolayout was the way to go...

Reply:
> Is there a preferred way to do this?

Not really. The preferred way is probably whatever is simplest/easiest.

> I thought Autolayout was the way to go

Using the autoresizing mask is still using Auto Layout when `translatesAutoresizingMaskIntoConstraints` is enabled. The system takes the value of the autoresizing mask and generates the appropriate constraints automagically.

To be clear: There isn't a right way and a wrong way. What you have is just fine and either approach should produce the exact same constraints. The `autoresizingMask` route might just be a bit simpler.

Reply:
Ok, thanks.  I'm going to leave it as is because, during testing, I've needed to modify the size of one or both of the subViews and having the constraints spelled out makes that easier.